### ISSUE TITLE
RPN loss very high?

### ISSUE THREAD

**Issue by ruiyuanlu** [facebookresearch/maskrcnn-benchmark#1019]
## ❓ Questions and Help
Hi guys, thanks for this generalized object detection benchmark, but I met a very strange issue that sometimes rpn loss is very high (around 4-20) and not decrease during training while other losses is normal (around 0.001-0.01). I've checked many times to make sure it is not because of the `requires_grad=False`, the code of `rpn` is directly cloned from the master branch. I've tried to increase/decrease base learning rate, but it didn't work. Increase learning rate results in loss to be nan, while decrease it didn't reduce the rpn loss.

Also, this condition occured randomly, I haven't found a stable way to reproduce.

Any tips would be very helpful, thanks~

**Comment by rnsandeep** [comment]
It is generally much harder to explain one instance of high loss. Better way is to visualize the predicted rpn boxes and your ground truth boxes. For that you need to keep track of which image you are training you can get it printed while running. Debug with visualization. BoxList class has many utility functions which allows you to draw the boxes easily.
